src/Systems/SpawnerSystem.cpp

```diff
@@ -48,63 +39,64 @@ void SpawnerSystem::processSpawner(SpawnerComponent& spawnerComponent,
+        enemyToSpawn.collision_data().y_offset(), enemyToSpawn.collision_data().width(),
+        enemyToSpawn.collision_data().height(),
+    };
+    const ColliderComponent colliderComponent{collisionData};
 
-    const ColliderComponent colliderComponent{enemyConfig.collisionData};
-    TransformComponent transformComponent{spawnerTransformComponent};
+    //TODO temporary solution working perfectly
+    transformComponent.position.x -= colliderComponent.collision.x * configSingleton.GetConfig().gameScale - 26;
+    transformComponent.position.y -= colliderComponent.collision.y * configSingleton.GetConfig().gameScale - 5;
 
-    transformComponent.position.x -= colliderComponent.collision.x * configSingleton.GetConfig().gameScale;
-    transformComponent.position.y -= colliderComponent.collision.y * configSingleton.GetConfig().gameScale;
+    TileComponent tileComponent{enemyToSpawn.texture_data().tile_id(), enemyToSpawn.texture_data().tile_set(),
+                                enemyToSpawn.texture_data().tile_layer()};
 
-    gCoordinator.addComponents(newMonsterEntity, enemyConfig.textureData, transformComponent, RenderComponent{},
-                               AnimationComponent{}, EnemyComponent{}, ColliderComponent{enemyConfig.collisionData},
-                               CharacterComponent{.hp = enemyConfig.hp});
+    gCoordinator.addComponents(newMonsterEntity, tileComponent, transformComponent, RenderComponent{},
+                               AnimationComponent{}, EnemyComponent{}, ColliderComponent{collisionData},
+                               CharacterComponent{.hp = static_cast<float>(enemyToSpawn.hp())});
 
     const Entity newEventEntity = gCoordinator.createEntity();
 
     auto newEvent = CreateBodyWithCollisionEvent(
         newMonsterEntity, "Enemy",
-        [&, newMonsterEntity, enemyConfig](const GameType::CollisionData& collisionData)
+        [&, newMonsterEntity, enemyToSpawn](const GameType::CollisionData& collision)
         {
-            if (!std::regex_match(collisionData.tag, config::playerRegexTag)) return;
+            // if (!std::regex_match(collision.tag, config::playerRegexTag)) return;
+            if (collision.entity != config::playerEntity) return;
 
-            if (!gCoordinator.hasComponent<CharacterComponent>(collisionData.entity) ||
-                gCoordinator.hasComponent<MultiplayerComponent>(collisionData.entity))
-                return;
-            auto& playerCharacterComponent{gCoordinator.getComponent<CharacterComponent>(collisionData.entity)};
+            if (!gCoordinator.hasComponent<CharacterComponent>(collision.entity)) return;
+            auto& playerCharacterComponent{gCoordinator.getComponent<CharacterComponent>(collision.entity)};
             playerCharacterComponent.attacked = true;
 
-            playerCharacterComponent.hp -= enemyConfig.damage;
+            playerCharacterComponent.hp -= enemyToSpawn.damage();
```

Comment:
czemu to ma ()? jak to jest pole klasy?

Reply:
Jest to element obiektu enemyToSpawn, który jest typu comm:Enemy, oznacza to tyle że jest to struktura proto, tak się po prostu pobiera wartość z niej. Jest tu proto dlatego, że przeniosłem całe tworznie potworów na serwer, klient nie robi nic związanego z obsługą przeciwników (no oprócz zadawania obrażeń i aktualnie niszczenie ich, ale to też zaraz będzie kotntrolowane przez serwer).